src/Compatibility/Microsoft.DotNet.PackageValidation/Package.cs

```diff
@@ -0,0 +1,178 @@
+
+        public IEnumerable<ContentItem> LibAssets { get; private set; }
+        
+        public IEnumerable<ContentItem> PackageAssets { get; private set; }
+
+        public IEnumerable<ContentItem> RuntimeSpecificAssets { get; private set; }
+
+        public IEnumerable<ContentItem> RuntimeAssets { get; private set; }
+
+        public bool HasRefAssemblies => RefAssets.Any();
+
+        public IEnumerable<string> Rids { get; private set; }
+
+        public IEnumerable<NuGetFramework> FrameworksInPackage { get; private set; }
+
+        public ContentItem FindBestRuntimeAssetForFramework(NuGetFramework framework)
+        {
+            SelectionCriteria managedCriteria = _conventions.Criteria.ForFramework(framework);
+            return _packageAssets.FindBestItemGroup(managedCriteria,
+                _conventions.Patterns.RuntimeAssemblies)?.Items.FirstOrDefault();
+        }
+
+        public ContentItem FindBestRuntimeAssetForFrameworkAndRuntime(NuGetFramework framework, string rid)
+        {
+            SelectionCriteria managedCriteria = _conventions.Criteria.ForFrameworkAndRuntime(framework, rid);
+            return _packageAssets.FindBestItemGroup(managedCriteria,
+                _conventions.Patterns.RuntimeAssemblies)?.Items.FirstOrDefault();
+        }
+
+        public ContentItem FindBestCompileAssetForFramework(NuGetFramework framework)
+        {
+            SelectionCriteria managedCriteria = _conventions.Criteria.ForFramework(framework);
+            if (RefAssets.Any())
+            {
+                return _packageAssets.FindBestItemGroup(managedCriteria,
+                    _conventions.Patterns.CompileRefAssemblies)?.Items.FirstOrDefault(); ;
+            }
+            else
+            {
+                return _packageAssets.FindBestItemGroup(managedCriteria,
+                    _conventions.Patterns.CompileLibAssemblies)?.Items.FirstOrDefault(); ;
+                
+            }
+        }
+
+        private int VersionComparer(int major, int minor, int patch) => major * 100 + minor * 10 + patch;
+
+        public async Task<MemoryStream> DownloadLatestStableVersionAsync(string nugetFeed = _nugetFeed)
+        {
+            SourceRepository repository = Repository.Factory.GetCoreV3(nugetFeed);
+            FindPackageByIdResource resource = await repository.GetResourceAsync<FindPackageByIdResource>();
+            SourceCacheContext cache = new SourceCacheContext();
+            IEnumerable<NuGetVersion> versions = await resource.GetAllVersionsAsync(PackageId, cache, NullLogger.Instance, CancellationToken.None);
+
+            if (versions == null)
+            {
+                throw new ArgumentException($"There is no version of the package in the feed {nugetFeed}");
+            }
+
+            NuGetVersion packageVersion = versions.Where(t => !t.IsPrerelease).OrderByDescending(t => VersionComparer(t.Major, t.Minor, t.Patch)).FirstOrDefault();
```

Comment:
I believe for servicing purposes we need to be able to not pick the very latest available package, but the latest one in the current band, i.e. 6.0.X during 6.0 servicing and not 7.0.X during 6.0 servicing.

Reply:
I think this comment is still valid. How do we compare against not latest but latest in the current band?

Reply:
I added the check to pick up the latest stable version than the current package version being built.